### PR TITLE
Fix for improper highlighting within the ternary operator

### PR DESCRIPTION
**Pull request by zoo1**
Now supports function and member access tokenizing. add corresponding specs

fix for #178 

Before:
ternbefore

After:
ternafter

### CODE REVIEW

**Comment by thomasjo** [comment]
@zoo1 Does this handle the pointer dereferencing scenario as well? Would be awesome to have a test added for that at least; e.g.

```c
a ? b->c : d
```

**Comment by zoo1** [comment]
@thomasjo it now does! A spec has been added to cover the functionality.

**Comment by thomasjo** [comment]
@zoo1 Beautiful stuff! Thanks 🙇

File: grammars/c.cson [hunk 1]
```diff
@@ -826,8 +826,17 @@
             'name': 'keyword.operator.ternary.c'
         'patterns': [
           {
+            'include': '#access'
+          }
+          {
             'include': '$self'
           }
+          {
+            'include': '#libc'
```
Comment (50Wliu) [inline grammars/c.cson]: Can these two be moved above `$self`?  I think generally `$self` should come last in patterns as sort of a default fallback.

Comment (zoo1) [inline grammars/c.cson]: @50Wliu 👍

**Pull request merged**